mypy/checkmember.py

```diff
@@ -670,23 +707,28 @@ def A(Generic(T)):
         @classmethod
         def foo(cls: Type[Q]) -> Tuple[T, Q]: ...
 
-    class B(A): pass
+    class B(A[str]): pass
 
     B.foo()
 
     original_type is the value of the type B in the expression B.foo()
     """
     # TODO: verify consistency between Q and T
     info = itype.type  # type: TypeInfo
+    if is_classmethod:
+        assert isuper is not None
+        t = expand_type_by_instance(t, isuper)
+    ids = {t.id for t in itype.args if isinstance(t, TypeVarType)}
```

Comment:
Could you add some comments about what is going on here?

Reply:
I added a detailed comment (and even found a little bug while writing it)/

Reply:
I double-checked that everything is OK with internal code bases.